### PR TITLE
[PyOV] Remove obsolete requirements

### PR DESCRIPTION
**Pull request by akuporos**
### Details
 - clean up  `compatibility/requirements-dev.txt` -- move reqs from it to `requirements_test.txt` as they need only for testing and left `cython` as it need for building old python api

### Tickets:
 - 101735

### CODE REVIEW

**Review by p-wysocki — APPROVED** [review]
LGTM, I hope removing test packages from compatibility doesn't require modifying CI pipelines :)

**Review by ilya-lavrenov — APPROVED** [review]

File: src/bindings/python/requirements_test.txt [hunk 1]
```diff
@@ -24,12 +24,14 @@ flake8_commas
 flake8_pep3101
 flake8_quotes
 mypy
-onnx==1.12.0
+onnx==1.12.0; platform_system == "Linux" and platform_machine == armv7l or platform_system == "Linux" and platform_machine == aarch64
```
Comment (ilya-lavrenov) [inline src/bindings/python/requirements_test.txt:27]: why do we need it? I suppose this version is available for each OS

Comment (akuporos) [reply src/bindings/python/requirements_test.txt]: condition here is wrong.

File: src/bindings/python/requirements_test.txt [hunk 2]
```diff
@@ -24,12 +24,14 @@ flake8_commas
 flake8_pep3101
 flake8_quotes
 mypy
-onnx==1.12.0
+onnx==1.12.0; platform_system != "Linux" and platform_machine != armv7l or platform_system != "Linux" and platform_machine != aarch64
```
Comment (ilya-lavrenov) [inline src/bindings/python/requirements_test.txt:27]: why ONNX tests are part of Python API tests at all?

Comment (p-wysocki) [reply src/bindings/python/requirements_test.txt:27]: It has been previously discussed in https://github.com/openvinotoolkit/openvino/pull/14985, I believe @mbencer agreed to consider managing/applying the move of ONNX test into ONNX Frontend, as it won't be a trivial change.

File: src/bindings/python/requirements_test.txt [hunk 3]
```diff
@@ -24,12 +24,15 @@ flake8_commas
 flake8_pep3101
 flake8_quotes
 mypy
-onnx==1.12.0
+onnx==1.12.0; platform_system != "Linux" and platform_machine != armv7l
+onnx==1.12.0; platform_system != "Linux" and platform_machine != aarch64
```
Comment (ilya-lavrenov) [inline src/bindings/python/requirements_test.txt:28]: looks like if we combine 2 lines, we can install onnx on any platform

Comment (akuporos) [reply src/bindings/python/requirements_test.txt]: updated

**Pull request merged**